### DISCUSSION TITLE
is there a way to identify the tenant by using Header instead of sub-domain or sub folder.

### DISCUSSION THREAD

**Discussion by praburocking** [django-tenants/django-tenants#611] [General]
Hi, is there a way to identify the tenant using the request Header like X-USER-ID=123245, instead of using sub-domain or sub folder?

**Comment by additionsolutions** [comment]
Yes, Its possible with little hack. Very similar to what is done in **'django_tenant_schemas'**

In my case I was passing 'schema_name' in request header.
Here is how I did it. 

1. You need **request** object in middleware to check value passed in request header. 
Hence we need to pass request for **get_tenant()** method.

Modify **main.py** in **TenantMainMiddleware** as follows:

        def process_request(self, request):
           ....
           try:
                    tenant = self.get_tenant(domain_model, hostname, request)
           ....
Note 'request' added as additional parameter, this will help us when we override get_tenant in our custom middleware

2. Add your custom middleware

        class XHeaderTenantMiddleware(TenantMainMiddleware):
            """
            Determines tenant by the value of the ``X-???`` in HTTP header.
            """
            def get_tenant(self, model, hostname, request):
               .....
               < Write your logic to get tenant based on value you passed in header using request.META.get() >
               .....
        
                domain = model.objects.select_related('tenant').get(tenant__schema_name=schema_name)
                return domain.tenant
In this example schema_name is used to get tenant.

Do not forgot to mention your custom middleware in **settings.py**

Hope this helps.

**Caveat**: I have used this logic in production using **'django_tenant_schemas'** but not with **'django_tenants'**, so its not fully tested.
